meshapp/src/components/profile/profile-textfield.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from "react";
+import { TextField, Box } from "@mui/material";
+
+import EditIcon from "@mui/icons-material/Edit";
+import SaveIcon from "@mui/icons-material/Save";
+
+/**
+ * A React component that renders a text field with editing capabilities.
+ * Includes a character limit and an edit/save mode toggle.
+ *
+ * Used in the Profile page (src/profile/profile-page.tsx).
+ *
+ * @param props - Properties of the component
+ * @param {string} props.label - The label
+ * @param {string} props.placeholder - The placeholder text
+ * @param {string} props.text - The initial text content
+ * @param {number} props.charLimit - The max number of characters allowed
+ */
+const ProfileTextField = (props: {
+  label: string;
+  placeholder: string;
+  text: string;
+  charLimit: number;
+}) => {
+  const [text, setText] = useState(props.text);
+  const [editMode, setEditMode] = useState(false);
+
+  // Enforce developer-defined character limit
+  const handleTextChange = (event: any) => {
```

Comment:
Instead of using any I think using React.ChangeEvent<HTMLTextAreaElement> would be more clear.